packages/apollo-gateway/src/snapshotSerializers/astSerializer.ts

```diff
@@ -14,8 +16,101 @@ export default {
     _refs: Refs,
     _printer: any,
   ): string {
-    return print(value)
+    return print(remapInlineFragmentNodes(value))
       .trim()
+      .replace(/\n\n/g, '\n')
       .replace(/\n/g, '\n' + indentation);
   },
 } as Plugin;
+
+/**
+ * This function converts potential InlineFragmentNodes that WE created
+ * (defined in ../QueryPlan, not graphql-js) to GraphQL-js compliant AST nodes
+ * for the graphql-js printer to work with
+ *
+ * The arg type here SHOULD be (node: AstNode | SelectionNode (from ../QueryPlan)),
+ * but that breaks the graphql-js visitor, as it won't allow our redefiend
+ * SelectionNode to be passed in.
+ *
+ * Since our SelectionNode still has a `kind`, this will still functionally work
+ * at runtime to call the InlineFragment visitor defined below
+ *
+ * We have to cast the `fragmentNode as undefined` and then at the bottom though, since there's
```

Comment:
I'm a little confused by this sentence, and also think you meant `unknown`

Reply:
Left off some words here :)